components/collections/codepointtrie_builder/cpp/list_to_ucptrie.cpp

```diff
@@ -2,111 +2,95 @@
+    valueWidth = UCPTRIE_VALUE_BITS_32;
+  } else {
+    fprintf(stderr, "Expected '8', '16', or '32' for 4th argument. Got %d",
+            valueWidthInt);
+    return 1;
+  }
 
-    icu::ErrorCode status;
+  icu::ErrorCode status;
 
-    icu::LocalUMutableCPTriePointer builder(umutablecptrie_open(
-        defaultValue,
-        errorValue,
-        status));
-    if (status.isFailure()) {
-        std::cerr << status.errorName() << std::endl;
-        return 1;
-    }
+  icu::LocalUMutableCPTriePointer builder(
+      umutablecptrie_open(defaultValue, errorValue, status));
+  if (status.isFailure()) {
+    fprintf(stderr, "LocalUMutableCPTriePointer builder failed: %s",
+            status.errorName());
+    return 1;
+  }
 
-    UChar32 cp = 0;
-    int32_t n = read_int32_t();
-    int32_t value = 0;
-    while(n--) {
-        value = read_int32_t();
-        umutablecptrie_set(builder.getAlias(), cp, value, status);
-        if (status.isFailure()) {
-            std::cerr << status.errorName() << std::endl;
-            return 1;
-        }
-        cp++;
-    }
-    std::cerr << "Saved values for code points up to " << cp << std::endl;
-
-    icu::LocalUCPTriePointer utrie(umutablecptrie_buildImmutable(
-        builder.getAlias(),
-        trieType,
-        valueWidth,
-        status));
+  for (int i = 0; i < n; ++i) {
+    umutablecptrie_set(builder.getAlias(), i, values[i], status);
     if (status.isFailure()) {
-        std::cerr << status.errorName() << std::endl;
-        return 1;
+      fprintf(stderr, "Unable to set %d to value: %d", i, values[i]);
+      return 1;
     }
+  }
+  icu::LocalUCPTriePointer utrie(umutablecptrie_buildImmutable(
+      builder.getAlias(), trieType, valueWidth, status));
+  if (status.isFailure()) {
+    fprintf(stderr,
+            "LocalUCPTriePointer umutablecptrie_buildImmutable failed: %s",
+            status.errorName());
+    return 1;
+  }
 
-    // Note: writesrc.h appears to only print LF, not CRLF
-    std::cout << "# This file is part of ICU4X. For terms of use, please see the file\n";
-    std::cout << "# called LICENSE at the top level of the ICU4X source tree\n";
-    std::cout << "# (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).\n";
-    std::cout << "\n";
-    std::cout << "# This file is auto-generated. Instructions: tools/list_to_ucptrie/README.md\n";
-    std::cout << "\n";
-    usrc_writeUCPTrie(stdout, "<unused>", utrie.getAlias(), UPRV_TARGET_SYNTAX_TOML);
-
-    return 0;
+  // Note: writesrc.h appears to only print LF, not CRLF
+  printf(
+      "# This file is part of ICU4X. For terms of use, please see the "
+      "file\n");
+  printf("# called LICENSE at the top level of the ICU4X source tree\n");
+  printf(
+      "# (online at: "
+      "https://github.com/unicode-org/icu4x/blob/main/LICENSE ).\n");
+  printf("\n");
+  printf(
+      "# This file is auto-generated. Instructions: "
+      "tools/list_to_ucptrie/README.md\n");
+  printf("\n");
```

Comment:
The license header is not needed now that this doesn't work as a binary anymore.

Reply:
Done.